Philips.CodeAnalysis.DuplicateCodeAnalyzer/AvoidDuplicateCodeAnalyzer.cs

```diff
@@ -67,65 +67,33 @@ public virtual HashSet<string> InitializeExceptions(ImmutableArray<AdditionalTex
 			return new HashSet<string>();
 		}
 
-		public virtual EditorConfigOptions InitializeEditorConfigOptions(ImmutableArray<AdditionalText> additionalFiles)
+		public virtual EditorConfigOptions InitializeEditorConfigOptions(AnalyzerOptions analyzerOptions, Compilation compilation)
 		{
 			EditorConfigOptions options = new EditorConfigOptions(DefaultDuplicateTokenThreshold);
+			var editorConfigHelper = new AdditionalFilesHelper(analyzerOptions, compilation);
 
-			foreach (AdditionalText additionalFile in additionalFiles)
-			{
-				string fileName = Path.GetFileName(additionalFile.Path);
-				StringComparer comparer = StringComparer.OrdinalIgnoreCase;
-				if (comparer.Equals(fileName, @".editorconfig"))
-				{
-					LoadEditorConfigOptions(options, additionalFile.GetText());
-				}
-			}
-
-			return options;
-		}
-
-		public virtual void LoadEditorConfigOptions(EditorConfigOptions options, SourceText text)
-		{
-			if (text == null)
-			{
-				return;
-			}
-			string diagnosticId = Helper.ToDiagnosticId(DiagnosticIds.AvoidDuplicateCode);
+			ExceptionsOptions exceptionsOptions = editorConfigHelper.LoadExceptionsOptions(Rule.Id);
+			options.IgnoreExceptionsFile = exceptionsOptions.IgnoreExceptionsFile;
+			options.GenerateExceptionsFile = exceptionsOptions.GenerateExceptionsFile;
 
-			foreach (TextLine textLine in text.Lines)
+			string strTokenCount = editorConfigHelper.GetValueFromEditorConfig(Rule.Id, @"token_count");
+			strTokenCount = strTokenCount.Trim();
+			try
 			{
-				string line = textLine.ToString();
-				if (line.Contains($@"dotnet_code_quality.{diagnosticId}.ignore_exceptions_file"))
+				int duplicateTokenThreshold = int.Parse(strTokenCount);

```

Comment:
int.TryParse (and then maybe you get rid of the try-catch?

Reply:
Replaced try/catch with tryparse here and in the new AvoidAssemblyChange analyzer, and added Diagnostics to show invalid editorconfig data to the user.